napari/layers/shapes/_shape_list.py

```diff
@@ -951,7 +951,7 @@ def outline(self, indices: Union[int, Sequence[int]]):
         triangles : np.ndarray
             Mx3 array of any indices of vertices for triangles of outline
         """
-        if isinstance(indices, int):
```

Comment:
I'm guessing there might be a perf reason, 
but can't we:
```
if not isinstance(indices, Sequence):
   indices = [indices]
```

and then have a branch less function ?

Reply:
That makes sense to me :+1: but what do you think @psobolewskiPhD @Czaki ?

Also, that would mean to change the current method definition to be something like:

```python
    def outline(self, indices: Union[int, Sequence[int]]):
        """Finds outlines of shapes listed in indices

        Parameters
        ----------
        indices : int | list
            Location in list of the shapes to be outline. If list must be a
            list of int

        Returns
        -------
        centers : np.ndarray
            Nx2 array of centers of outline
        offsets : np.ndarray
            Nx2 array of offsets of outline
        triangles : np.ndarray
            Mx3 array of any indices of vertices for triangles of outline
        """
        if not isinstance(indices, Sequence):
            indices = [indices]
        meshes = self._mesh.triangles_index
        triangle_indices = [
            i
            for i, x in enumerate(meshes)
            if x[0] in indices and x[1] == 1
        ]
        meshes = self._mesh.vertices_index
        vertices_indices = [
            i
            for i, x in enumerate(meshes)
            if x[0] in indices and x[1] == 1
        ]

        offsets = self._mesh.vertices_offsets[vertices_indices]
        centers = self._mesh.vertices_centers[vertices_indices]
        triangles = self._mesh.triangles[triangle_indices]

        t_ind = self._mesh.triangles_index[triangle_indices][:, 0]
        inds = self._mesh.vertices_index[vertices_indices][:, 0]
        starts = np.unique(inds, return_index=True)[1]
        for i, ind in enumerate(indices):
            inds = t_ind == ind
            adjust_index = starts[i] - vertices_indices[starts[i]]
            triangles[inds] = triangles[inds] + adjust_index

        return centers, offsets, triangles

```
?

Reply:
Certainly makes it simpler and easier to follow...

Reply:
I would even go one step further, introduce `def outlines(self,...)` that only take sequences, and have `def outline` with the only the `if not sequence...`. 

The reason being that `outline()` seem to be use in a single place, in a private function lower down the file, and from coverage report, it's not even clean the case where `indices` is an int is used.

We obviously can't remove it as it's public, but having a type stable method would be useful internally.

Reply:
Just to be sure, you mean doing something like:

```python
    def outline(self, indices: Union[int, Sequence[int]]):
        """Finds outlines of shapes listed in indices

        Parameters
        ----------
        indices : int | list
            Location in list of the shapes to be outline. If list must be a
            list of int

        Returns
        -------
        centers : np.ndarray
            Nx2 array of centers of outline
        offsets : np.ndarray
            Nx2 array of offsets of outline
        triangles : np.ndarray
            Mx3 array of any indices of vertices for triangles of outline
        """
        if not isinstance(indices, Sequence):
            indices = [indices]
        return self.outlines(indices)

    def outlines(self, indices: Sequence[int]):
        """Finds outlines of shapes listed in indices

        Parameters
        ----------
        indices : Sequence[int]
            Location in list of the shapes to be outline.

        Returns
        -------
        centers : np.ndarray
            Nx2 array of centers of outline
        offsets : np.ndarray
            Nx2 array of offsets of outline
        triangles : np.ndarray
            Mx3 array of any indices of vertices for triangles of outline
        """
        meshes = self._mesh.triangles_index
        triangle_indices = [
            i
            for i, x in enumerate(meshes)
            if x[0] in indices and x[1] == 1
        ]
        meshes = self._mesh.vertices_index
        vertices_indices = [
            i
            for i, x in enumerate(meshes)
            if x[0] in indices and x[1] == 1
        ]

        offsets = self._mesh.vertices_offsets[vertices_indices]
        centers = self._mesh.vertices_centers[vertices_indices]
        triangles = self._mesh.triangles[triangle_indices]

        t_ind = self._mesh.triangles_index[triangle_indices][:, 0]
        inds = self._mesh.vertices_index[vertices_indices][:, 0]
        starts = np.unique(inds, return_index=True)[1]
        for i, ind in enumerate(indices):
            inds = t_ind == ind
            adjust_index = starts[i] - vertices_indices[starts[i]]
            triangles[inds] = triangles[inds] + adjust_index

        return centers, offsets, triangles
```
right?

Reply:
Pushed a commit with the initial suggestion (a branchless function), but let me know what do you think about defining a a new method that accepts only `Sequence`

Reply:
> Just to be sure, you mean doing something like:
> 
> ```python
>     def outline(self, indices: Union[int, Sequence[int]]):
>         """Finds outlines of shapes listed in indices
> 
>         Parameters
>         ----------
>         indices : int | list
>             Location in list of the shapes to be outline. If list must be a
>             list of int
> 
>         Returns
>         -------
>         centers : np.ndarray
>             Nx2 array of centers of outline
>         offsets : np.ndarray
>             Nx2 array of offsets of outline
>         triangles : np.ndarray
>             Mx3 array of any indices of vertices for triangles of outline
>         """
>         if not isinstance(indices, Sequence):
>             indices = [indices]
>         return self.outlines(indices)
> 
>     def outlines(self, indices: Sequence[int]):
>         """Finds outlines of shapes listed in indices
> 
>         Parameters
>         ----------
>         indices : Sequence[int]
>             Location in list of the shapes to be outline.
> 
>         Returns
>         -------
>         centers : np.ndarray
>             Nx2 array of centers of outline
>         offsets : np.ndarray
>             Nx2 array of offsets of outline
>         triangles : np.ndarray
>             Mx3 array of any indices of vertices for triangles of outline
>         """
>         meshes = self._mesh.triangles_index
>         triangle_indices = [
>             i
>             for i, x in enumerate(meshes)
>             if x[0] in indices and x[1] == 1
>         ]
>         meshes = self._mesh.vertices_index
>         vertices_indices = [
>             i
>             for i, x in enumerate(meshes)
>             if x[0] in indices and x[1] == 1
>         ]
> 
>         offsets = self._mesh.vertices_offsets[vertices_indices]
>         centers = self._mesh.vertices_centers[vertices_indices]
>         triangles = self._mesh.triangles[triangle_indices]
> 
>         t_ind = self._mesh.triangles_index[triangle_indices][:, 0]
>         inds = self._mesh.vertices_index[vertices_indices][:, 0]
>         starts = np.unique(inds, return_index=True)[1]
>         for i, ind in enumerate(indices):
>             inds = t_ind == ind
>             adjust_index = starts[i] - vertices_indices[starts[i]]
>             triangles[inds] = triangles[inds] + adjust_index
> 
>         return centers, offsets, triangles
> ```
> 
> right?

Yes that's what I meant, sorry I was at NF summit and not super active.

Reply:
And It's just a suggestion, core devs would have to pitch in/validate.

Reply:
The @Carreau suggestion looks good and provides a cleaner interface.